tests/integ_tests/metrics_tests.rs

```diff
@@ -651,6 +654,81 @@ async fn request_fail_codes_otel() {
     }
 }
 
+// Tests that rely on Prometheus running in a docker container need to set
+// the `DOCKER_PROMETHEUS_RUNNING` env variable (see `is_prometheus_running_in_docker()`).
+#[rstest::rstest]
+#[tokio::test]
+async fn test_docker_metrics_with_prometheus(#[values(true, false)] use_http: bool) {
+    if !is_prometheus_running_in_docker() {
+        return;
+    }
+    let (otel_collector_addr, otl_protocl) = match use_http {
+        true => ("http://localhost:4318/v1/metrics", OtlpProtocol::Http),
+        false => ("http://localhost:4317", OtlpProtocol::Grpc),
+    };
+    let test_uid = format!(
+        "test_{}_",
+        uuid::Uuid::new_v4().to_string().replace("-", "")
+    );
+
+    // Configure the OTLP exporter with HTTP
+    let opts = OtelCollectorOptionsBuilder::default()
+        .url(otel_collector_addr.parse().unwrap())
+        .protocol(otl_protocl)
+        .global_tags(HashMap::from([("test_id".to_string(), test_uid.clone())]))
+        .build()
+        .unwrap();
+    let exporter = Arc::new(build_otlp_metric_exporter(opts).unwrap());
+    let telemopts = TelemetryOptionsBuilder::default()
+        .metrics(exporter as Arc<dyn CoreMeter>)
+        .metric_prefix(test_uid.clone())
+        .build()
+        .unwrap();
+    let rt = CoreRuntime::new_assume_tokio(telemopts).unwrap();
+    let mut starter = CoreWfStarter::new_with_runtime("test_metrics_with_docker_prometheus", rt);
+    let worker = starter.get_worker().await;
+    starter.start_wf().await;
+
+    // Immediately finish the workflow
+    let task = worker.poll_workflow_activation().await.unwrap();
+    worker
+        .complete_workflow_activation(WorkflowActivationCompletion::from_cmd(
+            task.run_id,
+            CompleteWorkflowExecution { result: None }.into(),
+        ))
+        .await
+        .unwrap();
+
+    let client = starter.get_client().await;
+    client.list_namespaces().await.unwrap();
+
+    // Give Prometheus time to scrape metrics
+    tokio::time::sleep(std::time::Duration::from_secs(5)).await;
```

Comment:
I wonder if you can hit some endpoint to force a scrape, but, since this is only in CI it's fine if it takes a bit.

Reply:
Since this is only CI, is there any reason I shouldn't just set the scrape interval to 1s, and then this sleep can go down to 2s?

Reply:
Yeah, you can do that too, but I thought that was defined in the same config file that gets used by the non-ci version of the docker compose. If not, then great.